### PR TITLE
SUP-6396: Pin version of urllib3

### PR DESCRIPTION
**Pull request by skalish**
# ↪️ Pull Request
This PR pins the version of the transitive dependency `urllib3` (required by `requests`) to use versions before 2.0.0.

It also updates the tested versions of Python and upgrades some dev dependencies.

## ✔️ PR Todo
- [x] Update relevant [docs](https://github.com/Datatamer/tamr-client/tree/main/docs) + docstrings

### CODE REVIEW

File: .python-version [hunk 1]
```diff
@@ -1,4 +1,4 @@
-3.6.13
+
 3.7.10
 3.8.10
 3.9.5
```
Comment (clint-richardson) [inline .python-version]: do you not need to add a 3.10 here? I'm reviewing each commit so apologies if this is addressed in a later one

Comment (skalish) [reply .python-version]: This should be addressed in a later commit. These are not strictly necessary at all but do help point to specific patch versions for local development.

File: stubs/responses.pyi [hunk 2]
```diff
@@ -15,9 +15,6 @@ def add(
     status: Optional[int] = None,
     json: Optional[JsonDict] = None,
 ): ...
-
-T = TypeVar("T")
-
-def activate(T) -> T: ...
+def activate(Callable) -> Callable: ...
```
Comment (clint-richardson) [inline stubs/responses.pyi:18]: what is this change doing? Is it related to updating the versions of python and urllib?

File: README.md [hunk 3]
```diff
@@ -38,5 +38,4 @@ For more see the [official docs](https://tamr-client.readthedocs.io/en/stable/).
 
 ## Maintainers
```
Comment (clint-richardson) [inline README.md:40]: @nbateshaus would you be comfortable with us adding you to this list?

File: stubs/responses.pyi (hunk 2 above)
Comment (skalish) [reply stubs/responses.pyi:18]: When updating the version of the `mypy` typechecker, it started complaining about the previous type-hint.  We use this only as a decorator of test functions for handling response mocking. This satisfies it and also seems reasonable.

**Review by clint-richardson — APPROVED** [review]

**Pull request merged**